chain/network/src/peer_manager/tests/routing.rs

```diff
@@ -18,6 +18,263 @@ use rand::Rng as _;
 use std::collections::HashSet;
 use std::sync::Arc;
 
+// test ping in a two-node network
+#[tokio::test]
+async fn ping_simple() {
+    init_test_logger();
+    let mut rng = make_rng(921853233);
+    let rng = &mut rng;
+    let mut clock = time::FakeClock::default();
+    let chain = Arc::new(data::Chain::make(&mut clock, rng, 10));
+
+    tracing::info!(target:"test", "start two nodes");
+    let pm0 = start_pm(clock.clock(), TestDB::new(), chain.make_config(rng), chain.clone()).await;
+    let pm1 = start_pm(clock.clock(), TestDB::new(), chain.make_config(rng), chain.clone()).await;
+
+    let id0 = pm0.cfg.node_id();
+    let id1 = pm1.cfg.node_id();
+
+    pm0.connect_to(&pm1.peer_info()).await;
+
+    tracing::info!(target:"test", "wait for {id0} routing table");
+    pm0.wait_for_routing_table(&[(id1.clone(), vec![id1.clone()])]).await;
+
+    tracing::info!(target:"test", "send ping from {id0} to {id1}");
+    pm0.send_ping(0, id1.clone()).await;
+
+    tracing::info!(target:"test", "await ping at {id1}");
```

Comment:
you have a race condition here:
if ping is received before wait_for_ping calls self.events.from_now() the MessageProcessed(Ping) event will be missed. Same for pong. You have to capture the event stream of the receiver before sending Ping/Pong.